pkg/specs/parse_test.go

```diff
@@ -2,38 +2,36 @@ package specs
 
 import (
 	"fmt"
-	"github.com/stretchr/testify/assert"
-	baremetalspecv1 "github.com/weaveworks/wksctl/pkg/baremetal/v1alpha3"
 	"io/ioutil"
-	"k8s.io/client-go/kubernetes/scheme"
-	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
 	"strings"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
 
 const clusterMissingClusterDefinition = `
 apiVersion: "cluster.weave.works/v1alpha3"
 kind: "BareMetalCluster"
 metadata:
- name: example
+  name: example
 spec:
- user: "vagrant"
+  user: "vagrant"
 `
 
 const clusterMissingBareMetalClusterDefinition = `
 apiVersion: "cluster.x-k8s.io/v1alpha3"
 kind: Cluster
 metadata:
- name: example
+  name: example
 spec:
- clusterNetwork:
-   services:
-     cidrBlocks: ["10.96.0.0/12"]
-   pods:
-     cidrBlocks: ["192.168.0.0/16"]
-   infrastructureRef:
-     kind: BareMetalCluster
-     name: example
+  clusterNetwork:
+    services:
+      cidrBlocks: ["10.96.0.0/12"]
+    pods:
+      cidrBlocks: ["192.168.0.0/16"]
+  infrastructureRef:
+    kind: BareMetalCluster
+    name: example
```

Comment:
Are we really testing anything, if this was not noticed?

Reply:
Now we are, as we're using strict parsing. Before, we did non-strict parsing, so all of these just resulted in unset fields.

Reply:
Looking again, the example is named "clusterMissingBareMetalClusterDefinition" so probably valid to have this wrong.